src/net/sourceforge/kolmafia/request/GenericRequest.java

```diff
@@ -177,15 +184,19 @@ public static URL getSecureRoot() {
   }
 
   private static HttpClient getClient() {
-    if (GenericRequest.client != null) {
-      return client;
-    }
+    // Wrap this in a synchronized block to avoid multiple clients being built in a rare condition
+    synchronized (clientRequestsSent) {
+      if (GenericRequest.client != null) {
+        return client;
+      }
```

Comment:
You can shift the "!= null" part outside the synchronization.

Reply:
On a second note, does it really matter if multiple clients get built? We always return the one attached to the class for new requests, and creating a client is not expensive. With the reset, it would reset X requests earlier than it has to, but over 10k requests I don't see much impact.

But equally with "not null" moved synchronizing shouldn't slow down too much.

Reply:
Moving it outside the block should be harmless in almost all cases, though given it wouldn't be thread safe as client is set to null when reset() is called; It'd best to have the field declared volatile.

After all, http requests are not single thread.

Reply:
oh yeah, forgot it got reset -- mixed it up with singleton pattern.

This could still lead to a rare NPE, though? Hits the sync block here, confirms non-null, reset runs in another thread and sets the field to null, `getClient` returns null.